docs/remote/devcontainer-cli.md

```diff
@@ -126,13 +126,101 @@ These steps above are also provided in the CLI repo's [README](https://github.co
 
 If you'd like to use the dev container CLI in your CI/CD builds or test automation, you can find examples of GitHub Actions and Azure DevOps Tasks in the [devcontainers/ci](https://github.com/devcontainers/ci) repository.
 
+## Pre-building
+
+The `devcontainer build` command allows you to build quickly build dev container image following the same steps the Remote - Containers extension or GitHub Codespaces will. This is particularly useful when you want to pre-build a dev container image using a CI or DevOps product like GitHub Actions.
+
+`build` accepts a path to the folder containing a `.devcontainer` folder or `.devcontainer.json` file. For example, `devcontainer build --workspace-folder <my_repo>` will build the container image for `my_repo`.
+
+### Example of building and publishing an image
+
+For example, you may want to pre-build a number of images that you then reuse across multiple projects or repositories. To do so, follow these steps:
+
+1. [Create](/docs/editor/versioncontrol.md#initialize-a-repository) a source code repository.
+
+1. Create dev container configuration for each image you want to pre-build, customizing as you wish (including [dev container Features](#dev-container-features-preview)). For example, consider this devcontainer.json file:
+
+    ```json
+    {
+        "build": {
+            "dockerfile": "Dockerfile"
+        },
+        "features": {
+            "ghcr.io/devcontainers/features/docker-in-docker:1": {
+            "version": "latest"
+            }
+        }
+    }
+    ```
+
+1. Use the `devcontainer build` command to build the image and [push](https://docs.docker.com/engine/reference/commandline/push/) it to your image registry. See documentation for your image registry (like the [Azure Container Registry](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli), [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images), or [Docker Hub](https://docs.docker.com/engine/reference/commandline/push)) for information on image naming and additional steps like authentication.
+
+    ```bash
+    devcontainer build --workspace-folder <my_repo> --push true --image-name <my_image_name>:<optional_image_version>
+    ```
+
+1. You have a couple of options for projects that will use your pre-built image:
```

Comment:
@Chuxel The more I re-read this section, the more I wonder if it's clear. 

If a user is building and pushing their image for the first time, it feels like they'd just need to go reference their published image in future projects' dev container config, and they're good to go.

Is my understanding of the following options accurate?
- We could mention the above (all you need to do to use this prebuilt image in other projects is reference the image from your container registry, and you're good to go), then:
- If you'd like to reference / use this dev container in the same repo from which you're pre-building it (https://github.com/devcontainers/spec/issues/74), you have the other 2 options (secondary devcontainer.json or `cacheFrom`)

Or maybe I'm forgetting why we wouldn't mention the first bullet as an option?

Reply:
Yeah there's two scenarios here:

1. You're building an image that gets shared in a bunch of spots.  In that case, you can just reference the image.
2. You just want to have a pre-built version of the resulting image for the current repository. This is where `--cache-from` is an alternative.

We could mention both, you are right.

Reply:
Thanks for the confirmation and great insights! I added an explanation on both scenarios, as I think that'd help users getting started with this flow.